src/reporter.cc

```diff
@@ -165,7 +166,7 @@ void MetadataReporter::SendMetadata(
                  << "B, dropping; entry " << *metadata_entry;
       continue;
     }
-    if (total_size + size > limit_bytes) {
+    if (entries.size() > limit_count || total_size + size > limit_bytes) {
```

Comment:
I'm confused by this line. Seems like we're already violating these bounds no? Unless `SendMetadataEntry` does some special batching.

Reply:
you are right, there should be an equality here since the count does not include the latest entry. Updated.